spec/forms/applicants/previous_reference_form_spec.rb

```diff
@@ -0,0 +1,150 @@
+      end
+    end
+
+    context "when neither yes nor no is chosen for applied_previously question" do
+      let(:params) { { model: applicant, applied_previously: "" } }
+
+      it "is invalid" do
+        call_save
+        expect(instance).not_to be_valid
+      end
+
+      it "adds custom blank error message" do
+        call_save
+        error_messages = instance.errors.messages.values.flatten
+        expect(error_messages).to include("Select yes if your client has applied for civil legal aid before")
+      end
+    end
+
+    context "when yes is chosen but no previous_reference is provided" do
+      let(:params) { { model: applicant, applied_previously: "true", previous_reference: "" } }
+
+      it "is invalid" do
+        call_save
+        expect(instance).not_to be_valid
+      end
+
+      it "adds custom blank error message" do
+        call_save
+        error_messages = instance.errors.messages.values.flatten
+        expect(error_messages).to include("Enter the reference number for any previous application")
+      end
+    end
+
+    context "with yes chosen but invalid previous reference number provided" do
+      context "with too many numbers" do
+        let(:params) { { model: applicant, applied_previously: "true", previous_reference: "3234567890123" } }
+
+        it "is invalid" do
+          call_save
+          expect(instance).not_to be_valid
+        end
+
+        it "adds custom blank error message" do
+          call_save
+          error_messages = instance.errors.messages.values.flatten
+          expect(error_messages).to include("Enter the reference number in the correct format")
+        end
+      end
+
+      context "with not enough numbers" do
+        let(:params) { { model: applicant, applied_previously: "true", previous_reference: "3234567890" } }
+
+        it "is invalid" do
+          call_save
+          expect(instance).not_to be_valid
+        end
+
+        it "adds custom blank error message" do
+          call_save
+          error_messages = instance.errors.messages.values.flatten
+          expect(error_messages).to include("Enter the reference number in the correct format")
+        end
+      end
+
+      context "when a non-integer included" do # rubocop:disable RSpec/RepeatedExampleGroupBody
+        let(:params) { { model: applicant, applied_previously: "true", previous_reference: "32345678901F" } }
+
+        it "is invalid" do
+          call_save
+          expect(instance).not_to be_valid
+        end
+
+        it "adds custom blank error message" do
+          call_save
+          error_messages = instance.errors.messages.values.flatten
+          expect(error_messages).to include("Enter the reference number in the correct format")
+        end
+      end
+
+      context "when 1 or 3 is not at the beginning" do # rubocop:disable RSpec/RepeatedExampleGroupBody
+        let(:params) { { model: applicant, applied_previously: "true", previous_reference: "32345678901F" } }
```

Comment:
@skdejong  I would make this

let(:params) { { model: applicant, applied_previously: "true", previous_reference: "423456789012" } }

I think then you should be able to remove the # rubocop:disable RSpec/RepeatedExampleGroupBody